File: akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpProtocol.scala
/**
 * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.http.scaladsl.model

import akka.http.impl.util.{ SingletonValueRenderable, ObjectRegistry }
import akka.http.javadsl.{ model ⇒ jm }

/** The protocol of an HTTP message */
final case class HttpProtocol private[http] (override val value: String) extends jm.HttpProtocol with SingletonValueRenderable

object HttpProtocols extends ObjectRegistry[String, HttpProtocol] {
  private def register(p: HttpProtocol): HttpProtocol = register(p.value, p)

  val `HTTP/1.0` = register(HttpProtocol("HTTP/1.0"))
  val `HTTP/1.1` = register(HttpProtocol("HTTP/1.1"))
}

File: akka-http-core/src/main/java/akka/http/javadsl/model/HttpProtocol.java
/**
 * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.http.javadsl.model;

/**
 * Represents an Http protocol (currently only HTTP/1.0 or HTTP/1.1). See {@link HttpProtocols}
 * for the predefined constants for the supported protocols.
 *
 * @see HttpProtocols for convenience access to often used values.
 */
public abstract class HttpProtocol {
    /**
     * Returns the String representation of this protocol.
     */
    public abstract String value();
}
